src/main/scala/no/ndla/articleapi/service/ReadService.scala

```diff
@@ -25,6 +25,12 @@ trait ReadService {
   val readService: ReadService
 
   class ReadService {
+    def getInternalIdByExternalId(externalId: Long): Option[api.ArticleIdV2] =
+      articleRepository.getIdFromExternalId(externalId.toString()) match {
```

Comment:
Når man har en match, slik som her, som har en `case None => None` kan det ofte forenkles ved å bruke `map`:
`articleRepository.getIdFromExternalId(externalId.toString).map(api.ArticleIdV2)`.

Her er det også litt magi rundt `case class`er som gjør at man kan skrive navnet på en klasse inne i map, som egentlig tar en funksjon som argument: `...map(api.ArticleIdV2)`. Svaret er egentlig at case classer deklarerer en del funksjoner i bakgrunnen for oss (som vi kan overstyre og legge til i et [companion object](https://docs.scala-lang.org/tour/singleton-objects.html)). Det som egentlig skjer er at det blir kalt en `apply`-funksjon som case classen definerer i bakgrunnen.